hal/targets/hal/TARGET_STM/TARGET_STM32F0/gpio_irq_api.c

```diff
@@ -29,7 +29,13 @@
  */
 #include <stddef.h>
 #include "cmsis.h"
+
 #include "gpio_irq_api.h"
+
+#include "device.h"
```

Comment:
You shouldn't need to include 'device.h` anymore (same observation applies to the changes below).